File: AoC/Day2/CSharp/Day2Part1.cs
namespace AoC.Day2.CSharp;

public class Day2Part1
{
    private List<List<int>> workingList { get; set; }
    public Day2Part1(bool partOne)
    {
        if (partOne)
        {
            //- Part One. - What do you get if you add up all the results of the multiplications?
            PartOne();
        }
        else
        {
            //- Part Two. - What do you get if you add up all of the results of just the enabled multiplications?
            // PartTwo(); 
        }
    }

    private void PartOne()
    {
        
        int safeAmount = 0;
        int increasing = 0;
        CleanList();
    }

    private void CheckingList()
    {
        
    }
    
    private List<List<int>> CleanList()
    {
        // Seperated by Levels.
        var input = File.ReadAllText("../../../Day2/input.txt").Split("\n");
        var cleaned = new List<List<int>>();
        var temp = new List<int>();

        //Report
        foreach (var report in input)
        {
            //Level
            var levels = report.Split(" ");
            foreach (var level in levels)
            {
                temp.Add(int.Parse(level));
            }

            cleaned.Add(temp);
            temp = [];
        }

        return cleaned;
        // foreach (var VARIABLE in cleaned)
        // {
        // Console.WriteLine(String.Join(";", VARIABLE));
        // }
        // Console.WriteLine(temp.);
    }
}

File: AoC/Day2/Python/day2part1.py
reports = open("AoC/Day2/input.txt", "r").read().splitlines()


def run():
    """How many reports are safe?"""
    cleaned = []
    safe_amount = 0
    increasing = 0
    for levels in reports:
        cleaned.append(levels.split(" "))

    for level in cleaned:
        last_index = len(level) - 1
        for i in range(len(level)):
            if i == last_index:
                continue

            # Checking difference of index and one after.
            difference = abs(int(level[i]) - int(level[i + 1]))

            if difference > 3 or difference < 1:
                break

            # Check if Increasing / Decreasing
            if int(level[i]) > int(level[i + 1]):
                increasing -= 1
            else:
                increasing += 1

        if abs(increasing) == abs(len(level) - 1):
            safe = True
        else:
            safe = False

        increasing = 0
        if safe:
            safe_amount += 1
    print(f"Py | Day 2, Part 1 | Safe amount: {safe_amount}")
